src/api/catalogueItem.tsx

```diff
@@ -191,3 +194,74 @@ export const useEditCatalogueItem = (): UseMutationResult<
     }
   );
 };
+
+export const useMoveToCatalogueItem = (): UseMutationResult<
+  TransferState[],
+  AxiosError,
+  MoveToCatalogueItem
+> => {
+  const queryClient = useQueryClient();
+  return useMutation(async (MoveToCatalogueItem: MoveToCatalogueItem) => {
+    const transferStates: TransferState[] = [];
+    let hasSuccessfulEdit = false;
+
+    const targetLocationInfo = {
+      name: MoveToCatalogueItem.targetLocationCatalogueCategory?.name ?? 'Root',
+      id: MoveToCatalogueItem.targetLocationCatalogueCategory?.id ?? null,
+    };
+
+    const promises = MoveToCatalogueItem.catalogueItems.map(
+      async (item: EditCatalogueItem, index) => {
+        const { name, ...itemWithoutName } = item;
+
+        if (
+          MoveToCatalogueItem.selectedItems[index].catalogue_category_id ===
+          item.catalogue_category_id
+        ) {
+          const errorTransferState: TransferState = {
+            name: item.name ?? '',
+            message:
+              'The destination cannot be the same as the catalogue item itself',
+            state: 'error',
+          };
+          transferStates.push(errorTransferState);
+
+          return;
+        }
+
+        return editCatalogueItem(itemWithoutName)
+          .then((result) => {
+            const successTransferState: TransferState = {
+              name: result.name ?? '',
```

Comment:
```suggestion
              name: result.name,
```
When successful it should be defined.